src/data/subtype.lean

```diff
@@ -37,6 +37,13 @@ protected theorem forall' {q : ∀x, p x → Prop} :
 lemma ext_iff {a1 a2 : {x // p x}} : a1 = a2 ↔ (a1 : α) = (a2 : α) :=
 ⟨congr_arg _, subtype.ext⟩
 
+lemma ext_iff_heq (h' : p = q) {a1 : {x // p x}} {a2 : {x // q x}} :
```

Comment:
Shouldn't this be called `heq_iff_coe_eq`? And it can probably be `simp`.
```suggestion
@[simp] lemma heq_iff_coe_eq (h' : p = q) {a1 : {x // p x}} {a2 : {x // q x}} :
```

Reply:
Some discussion about this lemma in https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.60p.20.3D.20q.60.20or.20.60.E2.88.80.20x.2C.20p.20x.20.E2.86.94.20q.20x.60/near/214968381

Reply:
Renamed as suggested, also golfed to term mode